Tests/AWSSDKSwiftCoreTests/AWSClientTests.swift

```diff
@@ -406,6 +406,35 @@ class AWSClientTests: XCTestCase {
             XCTFail(error.localizedDescription)
         }
     }
+    
+    func testSignedClient() {
+        let input = E()
+        let client = AWSClient(
+            accessKeyId: "foo",
+            secretAccessKey: "bar",
+            region: .useast1,
+            service: "s3",
+            serviceProtocol: .restxml,
+            apiVersion: "2013-12-02",
+            middlewares: [],
+            httpClientProvider: .createNew
+        )
+
+        do {
+            let awsRequest = try client.createAWSRequest(
+                operation: "CopyObject",
+                path: "/",
+                httpMethod: "PUT",
+                input: input
+            )
+
+            let request: AWSHTTPRequest = awsRequest.createHTTPRequest(signer: try client.signer.wait())
+
+            XCTAssertNotNil(request.headers["Authorization"].first)
```

Comment:
if you just want to ensure no error is thrown, we (following @weissi's wisdom 😉) normally use a pattern like this: 

```swift
var request: AWSHTTPRequest?
XCTAssertNoThrow(request = awsRequest.createHTTPRequest(signer: try client.signer.wait()))
XCTAssertNotNil(request?.headers["Authorization"].first)
```

This saves us an indentation and ensures all lines in the test are always hit.

Reply:
sounds better this way. I just copied the way another test is written. I'll change that test too.